### PR TITLE
feat(gatsby-plugin-benchmark-reporting): allow arbitrary restart

### PR DESCRIPTION
**Pull request by pvdz**
This change allows the benchmark reporter to start benchmarking arbitarily after a previous run finished, without having to explicitly request it. This allows us to benchmark incremental builds more reliably, which don't necessarily trigger all the events.

### CODE REVIEW

File: packages/gatsby-plugin-benchmark-reporting/src/gatsby-node.js [hunk 1]
```diff
@@ -272,6 +275,21 @@ class BenchMeta {
...
+    benchMeta = new BenchMeta()
+    // This should be set in the gatsby-config of the site when enabling this plugin
+    reportInfo(
+      `gatsby-plugin-benchmark-reporting: Will post benchmark data to: ${
+        BENCHMARK_REPORTING_URL || `the CLI`
+      }`
+    )
+
+    benchMeta.markStart()
+    benchMeta.markDataPoint(lifecycle)
```
Comment (wardpeet) [inline packages/gatsby-plugin-benchmark-reporting/src/gatsby-node.js:289]: Doesn't this make it call any lifecycle twice for markDataPoint when it hasn't started?

File: packages/gatsby-plugin-benchmark-reporting/src/gatsby-node.js [hunk 2]
```diff
@@ -285,36 +303,36 @@ process.on(`exit`, () => {
...
   lastApi = api
-  // This should be set in the gatsby-config of the site when enabling this plugin
-  reportInfo(
-    `gatsby-plugin-benchmark-reporting: Will post benchmark data to: ${
-      BENCHMARK_REPORTING_URL || `the CLI`
-    }`
-  )
-
-  benchMeta.markStart()
+  init(`onPreInit`)
```
Comment (wardpeet) [inline packages/gatsby-plugin-benchmark-reporting/src/gatsby-node.js:308]:
```suggestion
  init(`preInit`)
```

File: packages/gatsby-plugin-benchmark-reporting/src/gatsby-node.js (hunk 1 above)
Comment (pvdz) [reply packages/gatsby-plugin-benchmark-reporting/src/gatsby-node.js:289]: Oh dang, you're right. Removed it

**Comment by pvdz** [comment]
It's kind of irrelevant for incremental builds. This timestamp is more an indicator of load in general. It should be stable but if this number is an outlier, then the whole run could be considered an outlier for reasons unrelated to the benchmark.

Fwiw, I think it's fine to keep it like this. Perhaps somebody else has a different thought on this?

**Comment by pvdz** [comment]
Published in gatsby-plugin-benchmark-reporting@0.2.2

File: packages/gatsby-plugin-benchmark-reporting/src/gatsby-node.js [hunk 3]
```diff
@@ -272,6 +275,20 @@ class BenchMeta {
   }
 }
 
+function init(lifecycle) {
```
Comment (muescha) [inline packages/gatsby-plugin-benchmark-reporting/src/gatsby-node.js:278]: lifecycle is never used in this function?

**Pull request merged**